docs/architecture/devops-for-aspnet-developers/actions-vs-pipelines.md

```diff
@@ -0,0 +1,118 @@
+
+> [!NOTE]
+> The term "pipelines" can also be referred to by several different interchangeable words: _pipeline_, _workflow_ and _build_ are common terms. In this article, references to _Azure Pipelines_ are referring to [YAML Pipelines](/azure/devops/pipelines/get-started/pipelines-get-started?view=azure-devops#define-pipelines-using-yaml-syntax), and not the older UI-based [Classic Pipelines](/azure/devops/pipelines/get-started/pipelines-get-started?view=azure-devops#define-pipelines-using-the-classic-interface).
+
+## Agents and runners
+
+Before you examine pipelines themselves, you should consider how these pipelines _execute_. Both GitHub Actions and Azure Pipelines are really _orchestration engines_: when a pipeline is triggered, the system finds an "agent" and tells the agent to execute the jobs defined in the pipeline file.
+
+Azure Pipelines run on _agents_. The agent is written in .NET, so it'll run wherever .NET can run: Windows, macOS and Linux. Agents can even run in containers. Agents are registered to a [pool](https://docs.microsoft.com/azure/devops/pipelines/agents/pools-queues?view=azure-devops) in Azure Pipelines or to a repository or organization in GitHub. Agents can be _hosted_ or _private_.
+
+GitHub Workflows execute on _runners_. The runner code is essentially a fork of the Azure Pipelines code, so it's very similar. It's also cross-platform and you can also use _hosted_ or _self-hosted_ runners.
+
+### Hosted agents and runners
+
+Hosted agents (Azure Pipelines) and hosted runners (GitHub) are agents that are spun up and managed by Azure DevOps or GitHub respectively. You don't need to maintain any build infrastructure. When a pipeline triggers that targets a hosted agent, an instance of the specified agent image is created. The job is run by the agent on the instance, and once the job completes, the instance is destroyed. The same applies for hosted runners running GitHub workflows.
+
+> [!NOTE]
+> The list of software installed on Azure Pipeline images is listed in [this repo](https://github.com/actions/virtual-environments/tree/master/images). You can select the platform folder and examine the *README.md* files. You can find information on GitHub hosted runners [here](https://docs.github.com/actions/reference/specifications-for-github-hosted-runners).
+
+### Private agents and self-hosted runners
+
+There are times when you can't utilize hosted images. For example, when you:
+
+- Require SDKs or other software that is not installed on the images.
+- Need to access resources that aren't public (such as an internal Sonarqube server or an internal Artifactory instance).
+- Need to deploy to private networks.
+- Need to install licenses for 3rd party software required for building your code.
+- Need more storage or memory than is provided to the hosted agent images.
+- Need more time than the maximum build time limit for hosted agents.
+
+> [!INFORMATION]
```

Comment:
```suggestion
> [!IMPORTANT]
```

No alert for `[!INFORMATION]` exists, please refer to [Alerts in the Contributor Guidance](https://docs.microsoft.com/en-us/contribute/markdown-reference#alerts-note-tip-important-caution-warning).